openwisp_monitoring/device/api/views.py

```diff
@@ -203,6 +205,74 @@ def _write(self, request, pk):
                 metric.write(client['mac'])
             if created:
                 self._create_clients_graph(metric)
+        if 'resources' not in data:
+            return
+        self._write_cpu(data, pk, ct)
```

Comment:
Improvement: In these internal methods we don't need to full `data` so we can just pass `data['resources']` as `resources`